### PR TITLE
Correct README examples and add enhancement 

### PR DESCRIPTION
**Pull request by yupwei68**
Fix :https://github.com/Azure/terraform-azurerm-loadbalancer/issues/24
https://github.com/Azure/terraform-azurerm-loadbalancer/issues/18

merge PR: https://github.com/Azure/terraform-azurerm-loadbalancer/pull/14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,66 +15,71 @@ characteristics:
...
     ssh = ["Tcp", "22"]
   }
 
-  "lb_port" {
-    http  = ["80", "Tcp", "80"]
-    https = ["443", "Tcp", "443"]
+  lb_port = {
+    http   = ["80", "Tcp", "80"]
+    https  = ["443", "Tcp", "443"]
+    https2 = ["1443", "Tcp", "1443", "/", "Https"]
```
Comment (mybayern1974) [inline README.md:71]: In variables.tf, the definition of "lb_port" is "...[frontend_port, protocol, backend_port]...", i.e. there are totally 3 parameters, while here the "https2" has 5 parameters. Is it a mismatch or any description needs to be changed in the variables.tf?

File: main.tf [hunk 2]
```diff
@@ -49,18 +48,19 @@ resource "azurerm_lb_nat_rule" "azlb" {
...
   count               = length(var.lb_port)
   name                = element(keys(var.lb_port), count.index)
-  resource_group_name = azurerm_resource_group.azlb.name
+  resource_group_name = data.azurerm_resource_group.azlb.name
   loadbalancer_id     = azurerm_lb.azlb.id
   protocol            = element(var.lb_port[element(keys(var.lb_port), count.index)], 1)
   port                = element(var.lb_port[element(keys(var.lb_port), count.index)], 2)
   interval_in_seconds = var.lb_probe_interval
   number_of_probes    = var.lb_probe_unhealthy_threshold
+  request_path        = lower(element(var.lb_port[element(keys(var.lb_port), count.index)], 3)) == "http" || lower(element(var.lb_port[element(keys(var.lb_port), count.index)], 3)) == "https" ? element(var.lb_port[element(keys(var.lb_port), count.index)], 4) : ""
```
Comment (mybayern1974) [inline main.tf:57]: In the [Azure LB TF doc](https://www.terraform.io/docs/providers/azurerm/r/lb_probe.html), the "request_path" refers to a URI. Seems the assigned value to request_path is not a URI type. Or did I misunderstand anything?

File: test/fixture/main.tf [hunk 3]
```diff
@@ -6,17 +6,45 @@ resource "random_id" "rg_name" {
...
 
   remote_port = {
     ssh = ["Tcp", "22"]
   }
 
   lb_port = {
-    http = ["80", "Tcp", "80"]
+    http   = ["80", "Tcp", "80"]
+    https  = ["443", "Tcp", "443"]
+    https2 = ["1443", "Tcp", "1443", "/", "Https"]
```
Comment (mybayern1974) [inline test/fixture/main.tf:30]: The same question as above about the mis-match between 3 parameters mentioned in doc and 5 parameters here

**Comment by mybayern1974** [comment]
@yupwei68 , thank you for making this PR. I have some questions and appreciate if you could help me understand. thanks.

File: README.md (hunk 1 above)
Comment (yupwei68) [reply README.md:71]: I have changed the corresponding description in variables.tf

File: main.tf (hunk 2 above)
Comment (yupwei68) [reply main.tf:57]: Yes. It's defined as a URI. But there is no validation restriction in `lb_probe_resource` and in service api. I have tested the example works fine. I assume `\` is acceptable for the service.

File: test/fixture/main.tf (hunk 3 above)
Comment (yupwei68) [reply test/fixture/main.tf:30]: On fact, I have changed the description in variables.tf.

**Comment by yupwei68** [comment]
Hi @mybayern1974 Thanks for your comments. I have replied in the comments, please continue reviewing.

**Comment by yupwei68** [comment]
Hi @mybayern1974 I found that the protocol type of `lb_rule` and `lb_probe` is different. While before, it applies the `lb_rule` protocol to `lb_probe`. Currently, I have separate the `lb_probe` from `lb_rule`. Please continue reviewing.

**Pull request merged**